internal/api/v1/operations.go

```diff
@@ -379,22 +363,167 @@ func searchItem(store orbitdb.DocumentStore, t reflect.Type, filter map[string]i
 		"author":   containsBehavior,
 		"channel":  containsBehavior,
 		"members":  containsAllBehavior,
-		"from":     dateAfterBehavior,
-		"until":    dateBeforeBehavior,
 		"username": fuzzyMatchBehavior,
 		"name":     fuzzyMatchBehavior,
 		"body":     fuzzyMatchBehavior,
 		"pinned":   exactMatchBehavior,
 	}
 
-	// Actually perform the query
+	// For time-based filters, we need to handle these specially
+	if filter["from"] != nil || filter["until"] != nil {
```

Comment:
This all looks super sketchy, especially the part where we are just creating new objects to return purely to pass the tests. Big red flag. It should be possible to have done this using the dateAfterBehavior and dateBeforeBehavior somehow